src/stemmablenote.ts

```diff
@@ -194,29 +194,27 @@ export abstract class StemmableNote extends Note {
 
   /** Gets the `y` value for the top modifiers at a specific `textLine`. */
   getYForTopText(textLine: number): number {
-    if (!this.stave) throw new Vex.RERR('NoStave', 'No stave attached to this note.');
     const extents = this.getStemExtents();
-    if (extents) {
+    if (this.hasStem()) {
```

Comment:
`if (!extents) throw new Vex.RERR('InvalidState', 'Stem does not have extents.');`

then we can avoid `extents?` and `??` below.

Reply:
Ok I have also moved `getStemExtents` inside the `if`.